src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -325,7 +333,14 @@ export class ProsemirrorAdapter {
         }
     };
 
-    private handleCancelLinkMenu = () => {
+    private handleCancelLinkMenu = (event: CustomEvent<string>) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        if (event.detail === TAB) {
+            return;
+        }
+
```

Comment:
another viable solution might be simply getting rid of `limel-menu-surface` and having `limel-text-editor-link-menu` directly in `limel-portal` as the only child. If we do so, we lose the inbuilt dismiss fuctionalities, like scrim click (clicking outside to dismiss) or pressing <kbd>esc</kbd> to dismiss. 

However, give how much things we have changed in this PR, I believe that's easier to just get rid of  `limel-menu-surface` here and re-implement these functionalities decently for the link menu instead.

Reply:
I'm willing to try different approaches, however if we remove the limel-menu-surface we would have to add a lot more code to get back that functionality.